pe/pe.py

```diff
@@ -404,6 +404,36 @@ def recurse_resources(node, parent_name):
             res.add_lines(dataless_resources)
             self.file_res.add_section(res)
 
+    # Inspired by https://github.com/CYB3RMX/Qu1cksc0pe/blob/086db196d2de289f0784ae4d8ee03f34bf10354b/Modules/winAnalyzer.py#L446
+    def check_number_of_functions(self):
+        if len(self.binary.imported_functions) + len(self.binary.exported_functions) < 20:
+            res = ResultSection("PE file contains a suspiciously low number of functions")
+            res.add_line(
+                f"There are {len(self.binary.imported_functions)} import and "
+                f"{len(self.binary.exported_functions)} export functions."
+            )
+            self.file_res.add_section(res)
+
+            # Inspired by https://github.com/viper-framework/viper-modules/blob/00ee6cd2b2ad4ed278279ca9e383e48bc23a2555/lief.py#L760
+            if len(self.binary.exported_functions) == 0:
+                ResultSection("No exported function found", parent=self.file_res)
+
+            # Inspired by https://github.com/viper-framework/viper-modules/blob/00ee6cd2b2ad4ed278279ca9e383e48bc23a2555/lief.py#L798
+            if len(self.binary.imported_functions) == 0:
+                ResultSection("No imported function found", parent=self.file_res)
+
+    # Inspired by https://github.com/viper-framework/viper-modules/blob/00ee6cd2b2ad4ed278279ca9e383e48bc23a2555/lief.py#L945
+    def check_data_directories(self):
+        # Inspired by https://github.com/viper-framework/viper-modules/blob/00ee6cd2b2ad4ed278279ca9e383e48bc23a2555/lief.py#L963
+        if not self.binary.data_directories:
+            ResultSection("No data directory found", parent=self.file_res)
+
+    # Inspired by https://github.com/viper-framework/viper-modules/blob/00ee6cd2b2ad4ed278279ca9e383e48bc23a2555/lief.py#L1075
+    def check_relocations(self):
```

Comment:
Can you rename add_optional(self) to add_relocations(self), and merge this function with the add_reocations() that would already exists? I think you could use self.binary.has_relocations instead of getting the full list, that wait it's never used if we do not request a deep_scan.
And I think this function was never called?

Reply:
https://github.com/CybercentreCanada/assemblyline-service-pe/pull/54/commits/1b27090256fc8798bfafdc6d643998ff01386d10
A lot of unittest files hit on this. Will have to run in PB to confirm that it is not worth scoring